source/Cosmos.System2/Graphics/SVGAIIScreen.cs

```diff
@@ -0,0 +1,195 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Cosmos.HAL.Drivers.PCI.Video;
+
+namespace Cosmos.System.Graphics
+{
+    public class SVGAIIScreen : Canvas
+    {
+        public VMWareSVGAII xSVGAIIDriver;
+        internal Debug.Kernel.Debugger mSVGAIIDebugger = new Debug.Kernel.Debugger("System", "SVGAIIScreen");
+
+        public SVGAIIScreen() : base()
+        {
+            mSVGAIIDebugger.SendInternal($"Creting new SVGAIIScreen with default mode {defaultGraphicMode}");
+
+            xSVGAIIDriver = new VMWareSVGAII();
+            xSVGAIIDriver.SetMode((uint)defaultGraphicMode.Columns, (uint)defaultGraphicMode.Rows, (uint)defaultGraphicMode.ColorDepth);
+        }
+
+        public SVGAIIScreen(Mode aMode) : base(aMode)
+        {
+            ThrowIfModeIsNotValid(aMode);
+
+            xSVGAIIDriver = new VMWareSVGAII();
+            xSVGAIIDriver.SetMode((uint)aMode.Columns, (uint)aMode.Rows, (uint)aMode.ColorDepth);
+        }
+
+        public override Mode Mode
+        {
+            get => mode;
+            set
+            {
+                mode = value;
+                SetGraphicsMode(mode);
+            }
+        }
+
+        public override void DrawPoint(Pen pen, int x, int y)
+        {
+            Color xColor = pen.Color;
+
+            mSVGAIIDebugger.SendInternal($"Drawing point to x:{x}, y:{y} with {xColor.Name} Color");
+            xSVGAIIDriver.SetPixel((uint)x, (uint)y, (uint)xColor.ToArgb());
+            mSVGAIIDebugger.SendInternal($"Done drawing point");
+            xSVGAIIDriver.Update(0, 0, (uint)mode.Columns, (uint)mode.Rows);
+        }
+
+        public override void DrawArray(Color[] colors, int x, int y, int width, int height)
+        {
+            throw new NotImplementedException();
+            //xSVGAIIDriver.
+        }
+        public override void DrawPoint(Pen pen, float x, float y)
+        {
+            //xSVGAIIDriver.
+            throw new NotImplementedException();
+        }
+
+        public override void DrawFilledRectangle(Pen pen, int x_start, int y_start, int width, int height)
+        {
+            xSVGAIIDriver.Fill((uint)x_start, (uint)y_start, (uint)width, (uint)height, (uint)pen.Color.ToArgb());
+        }
+
+        public override List<Mode> getAvailableModes()
+        {
+            return new List<Mode>
+            {
+            /* 16-bit Depth Resolutions*/
+
+                /* SD Resolutions */
+                new Mode(320, 200, ColorDepth.ColorDepth16),
+                new Mode(320, 240, ColorDepth.ColorDepth16),
```

Comment:
Regarding getAvailableModes() would be interesting to see if VmWare has a mode to query for this instead of hardcode the values.
This could be object of a successive PR if you prefer.

Reply:
The VMWare driver doesn't have a hardcoded set of available modes which the user can use, instead it supports an arbitrary range of resolutions depending on the size of the VM window

What may be doable is having a check to tell what size the VM window is at, and resize accordingly, this might be better left to an enhancements PR